Kernel/Devices/SerialDevice.cpp

```diff
@@ -81,18 +39,20 @@ String SerialDevice::device_name() const
 
 UNMAP_AFTER_INIT void SerialDevice::initialize()
 {
-    set_interrupts(false);
-    set_baud(Baud38400);
-    set_line_control(None, One, EightBits);
+    set_interrupts(true);
     set_fifo_control(EnableFIFO | ClearReceiveFIFO | ClearTransmitFIFO | TriggerLevel4);
-    set_modem_control(RequestToSend | DataTerminalReady);
+    set_modem_control(RequestToSend | DataTerminalReady); // FIXME: implement this
 }
 
 UNMAP_AFTER_INIT void SerialDevice::set_interrupts(bool interrupt_enable)
 {
     m_interrupt_enable = interrupt_enable;
-
-    m_base_addr.offset(1).out<u8>(interrupt_enable);
+    if (interrupt_enable)
+        enable_irq();
+    else
+        disable_irq();
+    dbgln("{}: Setting interrupts enabled={}", m_tty_name, interrupt_enable);
+    m_base_addr.offset(1).out<u8>(interrupt_enable ? 0x01 : 0x00);
```

Comment:
interrupt_enable is a bool, so that ternary is useless :)

Reply:
I think we should be explicit about which interrupts we enable (there are multiple bit flags). Will add an enum for it.